subsys/net/lib/zperf/zperf_shell.c

```diff
@@ -609,17 +658,32 @@ static void udp_upload_cb(enum zperf_status status,
 {
 	const struct shell *sh = user_data;
 
+#ifdef CONFIG_ZPERF_SESSION_PER_THREAD
+	struct session *ses = CONTAINER_OF(result, struct session, result);
+#endif /* CONFIG_ZPERF_SESSION_PER_THREAD */
+
 	switch (status) {
 	case ZPERF_SESSION_STARTED:
 		break;
 
 	case ZPERF_SESSION_FINISHED: {
-		shell_udp_upload_print_stats(sh, result);
+		shell_udp_upload_print_stats(sh, result, true);
+
+#ifdef CONFIG_ZPERF_SESSION_PER_THREAD
+		ses->in_progress = false;
```

Comment:
I just wonder, shoudn't session state be updated by the respective work handler (`tcp_upload_async_work()`) when finished? In case someone uses zperf API instead of shell.

Reply:
The `in_progress` is only used in zperf shell to determine what to print with `zperf jobs` command. I suppose we could set the variable in shell too. I can move the variable setting to there so we avoid setting these shell specific parameters in non-shell code.
I noticed also that we need to add test compilation to zperf sample for the new option which is not set by default.

Reply:
Hmm, I just realized that we do not yet know the session in the shell when calling `tcp_upload_async_work()`, I think it is simpler if we leave the `in_progress` as is atm. If someone is not using shell, the variable is basically no-op and does no harm.